gemini/use-cases/retrieval-augmented-generation/qna_using_query_routing/qna_using_query_routing.ipynb

```diff
@@ -0,0 +1,1008 @@
+{
```

Comment:
<p>Glad you're using batch processing for this!</p><p> </p><p>You can also potentially improve the embedding quality if you set the <code>task_type</code> specific to what the documents are used for <a href="https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings#api_changes_to_models_released_on_or_after_august_2023" target="_blank">https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings#api_changes_to_models_released_on_or_after_august_2023</a></p><p> </p><p>You put the task type in the JSONL metadata for batch processing.</p><p> </p>

<pre class="ql-syntax" spellcheck="false">{task_type": "RETRIEVAL_DOCUMENT", "title": "document title", "content": "I would like embeddings for this text!"}
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/493/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>